src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -562,16 +562,46 @@ impl CatalogState {
                     .map(|d| Datum::String(d))
                     .unwrap_or(Datum::Null);
                 let pgtype = mz_pgrepr::Type::from(&column_type.scalar_type);
+                let (type_name, type_oid) = match &column_type.scalar_type {
+                    ScalarType::List {
+                        custom_id: Some(custom_id),
+                        ..
+                    }
+                    | ScalarType::Map {
+                        custom_id: Some(custom_id),
+                        ..
+                    }
+                    | ScalarType::Record {
+                        custom_id: Some(custom_id),
+                        ..
+                    } => {
+                        let entry = self.get_entry(custom_id);
+                        // NOTE(benesch): the `mz_columns.type text` field is
```

Comment:
fwiw `regtype` should resolve the OID to the right entry

Reply:
Yeah, and you can join `mz_columns.type_oid` against `mz_types.oid` by hand! It's just at odds with the rest of the `mz_catalog` tables that you have to do the join with OID compatibility layer instead of being able to do it with Materialize's native global IDs.